File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValid(vector<vector<char>>& board, int row, int col, char ch){
        int n = board.size();
        int m = board[0].size();
        
        for(int p=0;p<m;p++){
            if(p!=col){
                if(ch==board[row][p]) return false;
            }
            if(p!=row){
                if(ch==board[p][col]) return false;
            }
        }
        int startRow = (row/3)*3;
        int startCol = (col/3)*3;
        for (int i = startRow; i < startRow + 3; i++) 
            for (int j = startCol; j < startCol + 3; j++) 
                if ((i != row || j != col) && board[i][j] == ch) 
                    return false;
        return true;
    }
    bool isValidSudoku(vector<vector<char>>& board) {
        int n=board.size(),m=board[0].size();
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                char ch = board[i][j];
                if(ch!='.'){
                    if(!isValid(board, i , j, ch)){
                        return false;
                    }
                }
            }
        }
        return true;
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        boolean[][] row = new boolean[9][9];
        boolean[][] col = new boolean[9][9];
        boolean[][] box = new boolean[9][9];
        for(int i=0;i<9;i++){
            for(int j=0;j<9;j++){
                if(board[i][j]=='.') continue;
                int digit = board[i][j] - '1';
                int boxIndex = (i/3) *3 + j/3;
                if(row[i][digit] || col[j][digit] || box[boxIndex][digit]){
                    return false;
                }
                row[i][digit] = true;
                col[j][digit] = true;
                box[boxIndex][digit] = true;
            }
        }
        return true;
    }
}